### ISSUE TITLE
Garbled up image using IPEX (Intel Extension for PyTorch)

### ISSUE THREAD

**Issue by 17314642** [Stability-AI/StableCascade#62]
Using notebook from `inference/text_to_image.ipynb` and replacing mentions of `cuda` to `xpu` made it possible to generate images on Intel Arc GPU but stage C images are garbled and are being upscaled either to black image or same garbled mess.

It is working on CPU.

![image](https://github.com/Stability-AI/StableCascade/assets/57731977/759dc0c6-cc50-470c-8892-fe5a864122ef)

**Comment by JohnnyRacer** [comment]
@17314642  Are the weights loaded in FP16 when using the GPU? There's a bug where the FP16 generation is bugged, on the CPU it's probably forced to use FP32 so that's why it works fine.

**Comment by 17314642** [comment]
> @17314642 Are the weights loaded in FP16 when using the GPU? There's a bug where the FP16 generation is bugged, on the CPU it's probably forced to use FP32 so that's why it works fine.


No, I'm using `stage_b_lite_bf16.safetensors` and `stage_c_lite_bf16.safetensors` using BFloat16 dtype.
